### ISSUE TITLE
CV-03. Проверить гипотезы по дисбалансу и ratio-aware training

### ISSUE THREAD

**Issue by mephiacc-david** [Bersserker/hakathon_room_cv#9]
Что нужно сделать: сравнить weighted CE, balanced sampler, ratio weighting, а при необходимости — обучение сначала на надежных данных, потом дообучение на всем наборе.
Артефакт: reports/ablation_loss_sampling_ratio_v1.csv → передает У1 для выбора рабочего рецепта.
Зависимости: baseline OOF от CV-02, ratio audit от У2.
DOR: baseline метрики зафиксированы; есть рекомендация по ratio.
DOD: минимум 4 сравнимых запуска на одном split version; в отчете есть разница с baseline и решение “берем / не берем”.
Оценка трудозатрат: 1.5 дн
Приоритет: P1

**Comment by mephiacc-david** [comment]
По коду проекта сейчас есть:

baseline модель: ConvNeXt-Tiny
train script: src/training/train_image.py
config: configs/model/image_baseline_v1.yaml
split: data/splits/splits_v1.json
baseline OOF: artifacts/oof/baseline_v1/
baseline метрика: reports/baseline_metrics_v1.md
ratio audit: reports/label_quality_ratio.md

Значит делать так.

──────────────────────────────────────────────────────────

CV-03: проверить loss / sampler / ratio

Цель: не новая модель. Берем тот же ConvNeXt-Tiny, тот же split, меняем только обучение.

Сейчас код умеет только:

criterion = nn.CrossEntropyLoss()
train_loader = build_loader(..., shuffle=True)

То есть сейчас есть только обычный CE + shuffle.

Нужно добавить в src/training/train_image.py:

Weighted CE
Добавить class weights по train fold:
nn.CrossEntropyLoss(weight=class_weights)
Balanced sampler
Вместо shuffle=True использовать WeightedRandomSampler, чтобы редкие классы чаще попадали в batch.
Ratio weighting

Сейчас RoomDataset возвращает:
img, label

Нужно, чтобы при ratio training возвращал:
img, label, sample_weight

Где:
sample_weight = clip(ratio, 0.75, 1.0)

Потом loss считать не сразу средним, а по объектам:
loss = CE(outputs, labels, reduction="none")
loss = (loss * sample_weight).mean()

Опционально reliable-first

Сначала обучать на:
ratio >= 1.0

или хотя бы:
ratio >= 0.75

Потом дообучать на всем train fold.

Но это optional. Минимум — 4 запуска.

──────────────────────────────────────────────────────────

Какие config сделать для CV-03

Создать отдельные yaml:

configs/model/cv03_baseline_ce.yaml
configs/model/cv03_weighted_ce.yaml
configs/model/cv03_balanced_sampler.yaml
configs/model/cv03_ratio_weighting.yaml

Все почти копии image_baseline_v1.yaml.

Менять только:

experiment:
loss: ce / weighted_ce / ratio_ce
sampler: shuffle / balanced
ratio_policy: none / clip_075_100

И разные output dirs:

artifacts:
oof_dir: artifacts/oof/cv03_weighted_ce
report_path: reports/cv03_weighted_ce.md

──────────────────────────────────────────────────────────

Как запускать CV-03

Все на одном split:

uv run python src/training/train_image.py --config configs/model/cv03_baseline_ce.yaml --all-folds
uv run python src/training/train_image.py --config configs/model/cv03_weighted_ce.yaml --all-folds
uv run python src/training/train_image.py --config configs/model/cv03_balanced_sampler.yaml --all-folds
uv run python src/training/train_image.py --config configs/model/cv03_ratio_weighting.yaml --all-folds

Потом собрать таблицу:

reports/ablation_loss_sampling_ratio_v1.csv

Колонки:

run_name, split_version, backbone, loss, sampler, ratio_policy, oof_macro_f1, shadow_macro_f1, delta_vs_baseline, decision

Baseline из проекта:

OOF macro_f1 = 0.622849
Shadow macro_f1 = 0.681315

Решение:

если лучше baseline стабильно =  берем
если хуже или равно = не берем
